### PR TITLE
Add PriceTest and ItemTest files and cases

### PR DESCRIPTION
**Pull request by Bahamas20**
New Testcases added for Price and Item 
All supplier components now covered for test cases
Item class now edited to take in strings and numbers but cannot take in numbers only

### CODE REVIEW

File: src/main/java/seedu/address/model/person/Price.java [hunk 1]
```diff
@@ -11,8 +11,9 @@ public class Price {
 
 
     public static final String MESSAGE_CONSTRAINTS =
-            "Price numbers should only contain numbers, and it should be in the $xx..x.xx format";
-    public static final String VALIDATION_REGEX = "\\$" + "\\d" + "\\." + "\\d{2}";
+            "Price numbers should only contain numbers, and it should be in 2 decimal places starting "
+                    + "starting with a $ sign";
+    public static final String VALIDATION_REGEX = "\\$" + "\\d+" + "\\." + "\\d{2}";
```
Comment (hauchongtang) [inline src/main/java/seedu/address/model/person/Price.java:16]: Nice constraint enforcement for 2 decimals for Price.

File: src/test/java/seedu/address/model/person/ItemTest.java [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    }
+
+    @Test
+    public void constructor_invalidItem_throwsIllegalArgumentException() {
+        String invalidItem = "";
+        assertThrows(IllegalArgumentException.class, () -> new Item(invalidItem));
+    }
+
+    @Test
+    public void isValidItem() {
```
Comment (hauchongtang) [inline src/test/java/seedu/address/model/person/ItemTest.java:23]: Quite a comprehensive enforcement of constraints here as well. Good use of assertions in test.

File: src/test/java/seedu/address/model/person/PriceTest.java [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    }
+
+    @Test
+    public void constructor_invalidPrice_throwsIllegalArgumentException() {
+        String invalidPrice = "";
+        assertThrows(IllegalArgumentException.class, () -> new Price(invalidPrice));
+    }
+
+    @Test
+    public void isValidPrice() {
```
Comment (weesiau) [inline src/test/java/seedu/address/model/person/PriceTest.java:23]: Comprehensive invalid examples

File: docs/index.md [hunk 4]
```diff
@@ -1,6 +1,6 @@
 ---
 layout: page
-title: AddressBook Level-3
+title: Salesy
```
Comment (hauchongtang) [inline docs/index.md:3]: Good catch !

**Review by hauchongtang — APPROVED** [review]
LGTM ! Test cases and enforcement of realistic constraints are well done.

**Pull request merged**